python/ml4ir/applications/ranking/tests/test_feature_fns.py

```diff
@@ -0,0 +1,80 @@
+import tensorflow as tf
+import numpy as np
+
+from ml4ir.applications.ranking.features.feature_fns import categorical
+from ml4ir.base.tests.test_base import RelevanceTestBase
+
+
+class FeatureLayerTest(RelevanceTestBase):
+
+    def test_sequence_categorical_vector_embedding(self):
+        """
+        Asserts the conversion of a categorical string tensor into a categorical embedding
+        Works by converting the string into indices using a vocabulary file and then
+        converting the indices into categorical embeddings
+
+        The embedding dimensions, buckets, etc are controlled by the feature_info
+        """
+        embedding_weights = np.random.randn(5 + 1, 32)
+        feature_info = {
```

Comment:
you can probably define this as a constant at the top of the file, and fill only what is test dependant ex: 
feature_info["feature_layer_info"]["args"]["output_mode"] = ""embedding"
feature_info["feature_layer_info"]["args"]["embeddings_initializer"] = tf.keras.initializers.Constant(embedding_weights)

feature_info["feature_layer_info"]["args"][""one_hot""]
 so that it makes it crystal clear what changes for each test, and avoid code/conf duplication.

Reply:
Let me do that.